src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -91,10 +91,35 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
         // tools for loading the compiled syntax tree from the dll resource (later setup for shipping Q# libraries)
 
         /// <summary>
-        /// Given a stream containing the binary representation of compiled Q# code, returns the corresponding Q# compilation.
-        /// Returns true if the compilation could be deserialized without throwing an exception, and false otherwise.
+        /// Given a binary representation of compiled Q# code, returns the corresponding Q# compilation.
+        /// Returns true if the compilation could be deserialized without throwing an exception, and it is not empty. False otherwise.
```

Comment:
I think the phrasing is a bit misleading; the method returns true even if the compilation has no namespaces but just an empty array. It only checks if that array can be accessed without throwing. 
```suggestion
        /// Returns true if the compilation could be deserialized without throwing an exception, and it is properly instantiated. False otherwise.
```